src/CBot.Modules.NewRelic/NewRelicMiddleware.cs

```diff
@@ -17,6 +17,12 @@ public class NewRelicMiddleware : IMiddleware
 
 		private readonly IManageNewRelic newRelicManager;
 
+		private static readonly string AllApplicationsCommand = $"{Parameters.NewRelic} applications";
+
+		private static readonly string AllApplicationsDetailCommand = $"{Parameters.NewRelic} all detail";
```

Comment:
This was previously `newrelic applications detail` but since we have the optional account parameter now, we can't use the ExactMatchHandler since both this and the one above start with `newrelic applications`, so I changed this to `newrelic all detail`. Opn to other suggestions/wording

Reply:
I wonder if the RegEx handler would help in this scenario?

Reply:
I'll look into this

Reply:
I fiddled around with regex for a bit, but wasn't quite able to come up with something that would allow keeping this as `applications`.

I tried something like:
`newrelic applications($|[ ]+[\w]*)` - looks for "newrelic applications" then either ends there (`$`), or looks for one or more spaces (`[ ]+`) followed by one or more word characters (`[\w]+`). The problem with this is that it would also match "detail" as the account name, and it seems like regex is not great at _not_ matching a certain word (ie don't match if next word is "detail"). I'm no regex whiz though, so maybe this is doable?

Reply:
After some more googling/fiddling, I think we could use:
`^newrelic applications(|[ ]+((?!detail).)+[\w]+)$`
`^newrelic applications detail(|[ ]+[\w]+)$`
for
`newrelic applications`
`newrelic applications detail`

However, this seems harder to read (I didn't know `?!` was a "negative look-ahead" that could be used for this...). It also sort of creates a dependency between the two commands, as the first regex now contains the word "detail" which only previously existed as part of the `AllApplicationsDetailCommand`

Reply:
I"m pretty handy with regex -
I am not sure what the exact scenarios you're looking to match are

To match either:
"newrelic applications"
or
"newrelic applications detail"

I'd use: "^newrelic applications(?:\s+detail)?$"
https://regex101.com/r/OZlZIY/1

Let me know if that's not the exact string you're looking to match (seems like maybe there's a word after detail?)

Reply:
@scott-goodman would that would what Eric suggested?

Reply:
Just to clarify for @protoET, we're looking to match two different things with different handlers. We want to match:
`newrelic applications account`
`newrelic applications detail account`
where `account` is optional and each scenario calls a different method. The regex Eric posted would match both (without account), but we only want to match one so we need two patterns. The ones I posted above seem to do the job, I'm just concerned about readability but maybe that's just an issue of me not being super comfortable with regex...

https://regex101.com/r/OZlZIY/4
https://regex101.com/r/OZlZIY/5

Reply:
I just noticed another problem with using `RegexHandler`:
![image](https://user-images.githubusercontent.com/49035355/55427442-a2e06080-554c-11e9-8a0d-6f761dbcf869.png)
Not very user-friendly help...

Reply:
gotcha,

I might suggest for the non-detail:
`^newrelic applications(?:\s+(?!detail).\w+)?$`
https://regex101.com/r/OZlZIY/6

& detail:
`^newrelic applications detail(?:\s+\w+)?$`
https://regex101.com/r/OZlZIY/7

Reply:
Updated. I also found a fix to the help text issue - the `RegexHandler.For(string pattern)` doesn't allow you to specify help text so it just spat the regex back out, but by using the actual `RegexHandler` constructor you can specify the help text. Looks nice and friendly now: 
![image](https://user-images.githubusercontent.com/49035355/55430048-ac6cc700-5552-11e9-88a5-f818422e8e54.png)